listelab-contrato/Controllers/ListaController.cs

```diff
@@ -14,21 +16,48 @@ namespace ListElab.Contrato.Controllers
     [ApiController]
     public class ListaController : ControladorPadrao<ListaQuestoes, IServicoListaQuestoes, DtoListaQuestoes>
     {
-        /// <summary>
-        /// Retorna uma lista de questoes discursivas
-        /// </summary>
-        /// <param name="filtro">O filtro de questão.</param>
-        /// <returns>Retorna objeto de resposta de sucesso ou falha, contendo o objeto desejado, caso sucesso.</returns>
-        [HttpPost]
-        [Route("filtro")]
-        [Authorize]
-        public ActionResult<DtoResultado<DtoListaQuestoes>> ConsulteDiscursivasComFiltro([FromBody] FiltroQuestao filtro)
-        {
-            return ExecuteAcaoAutorizada(() =>
-            {
-                var resultado = Servico().ConsulteQuestoes(filtro);
-                return Ok(DtoResultado<DtoListaQuestoes>.ObtenhaResultado(resultado, "Consulta realizada sem erros"));
-            });
+        /// <summary>
+        /// Filtro de lista. Funciona de forma cumulativa. Se eu escolher nivel de dificuldade 'Fácil' e Tempo Máximo de Resposta 10 minutos, então 
+        /// o a api só retornar as listas que atendem aos dois filtros. Não é preciso preencher todos os filtros, somente àqueles que se deseja usar.
+        /// </summary>
+        /// <param name="nivelDificuldade">1 - 'Muito Fácil'; 2 - 'Fácil', 3 - 'Médio', 4 - 'Difícil', 5 - 'Muito Difícil'</param>
+        /// <param name="disciplina">Deve-se passar o código da disciplina desejada. O filtro busca por listas que possuem questões dessa disciplina.</param>
+        /// <param name="areaDeConhecimento">Deve-se passar o código da área de conhecimento desejada. O filtro busca por listas que possuem questões dessa área de conhecimento.</param>
+        /// <param name="tempoEsperadoResposta">Tempo máximo de resposta da lista expresso em minutos.</param>
+        /// <param name="usuario">Autor da lista.</param>
+        /// <param name="id">O id da lista, caso precise só de uma.</param>
+        /// <param name="tags">Busca as tags das questões e traz as listas que possuem questões com as tags passada.</param>
+        /// <returns></returns>
+        [HttpGet]
+        [Authorize]
+        [Route("filtro")]
+        public ActionResult<DtoResultado<DtoListaQuestoes>> ConsulteComFiltro(

```

Comment:
Realmente há necessidade de passar todos esses parâmetros? Por que não passar um objeto filtro?

Reply:
Antes era assim mesmo, a gente passava o objeto filtro. O problema é que pediram para que isso fosse passado no parâmetro da requisição para facilitar. E também pesquisando sobre boas práticas para construção de api foi identificado que esse é o padrão para filtros.

https://www.meziantou.net/how-to-document-an-asp-net-core-web-api-using-openapi-specification-swagger.htm